src/formElementPlugins/listBuilder/src/ApplicationView.tsx

```diff
@@ -0,0 +1,339 @@
+  isValid?: boolean
+  value: ResponseFull
+}
+
+type ListItem = { [code: string]: InputResponseField }
+
+const ApplicationView: React.FC<ApplicationViewProps> = ({
+  element,
+  parameters,
+  onSave,
+  onUpdate,
+  Markdown,
+  initialValue,
+  currentResponse,
+  applicationData,
+  allResponses,
+}) => {
+  const { isEditable } = element
+  const {
+    label,
+    description,
+    createModalButtonText = strings.BUTTON_LAUNCH_MODAL,
+    modalText,
+    addButtonText = strings.BUTTON_ADD,
+    updateButtonText = strings.BUTTON_UPDATE,
+    deleteItemText = strings.BUTTON_DELETE,
+    inputFields,
+    displayFormat = getDefaultDisplayFormat(inputFields),
+    displayType = DisplayType.CARDS,
+  } = parameters
+
+  const [currentInputResponses, setCurrentInputResponses] = useState<ListItem>(
+    resetCurrentResponses(inputFields)
+  )
+
+  const [listItems, setListItems] = useState<ListItem[]>(initialValue?.list ?? [])
+  const [selectedListItem, setSelectedListItem] = useState<number | null>(null)
+  const [open, setOpen] = useState(false)
+  const [inputError, setInputError] = useState(false)
+
+  useEffect(() => {
+    onSave({
+      text: listItems.length > 0 ? createTextString(listItems, inputFields) : undefined,
+      list: listItems,
+    })
+  }, [listItems])
+
+  // This is sent to ApplicationViewWrapper and runs instead of the default responseMutation
+  const innerElementUpdate =
+    (code: string) =>
+    async ({ variables: response }: { variables: InputResponseField }) => {
+      const newResponses = { ...currentInputResponses, [code]: response }
+      setCurrentInputResponses(newResponses)
+      if (!anyErrorItems(newResponses, inputFields)) setInputError(false)
+    }
+
+  const updateList = async () => {
+    if (anyErrorItems(currentInputResponses, inputFields)) {
+      setInputError(true)
+      return
+    }
+    // Add item
+    if (selectedListItem === null) {
+      setListItems([...listItems, currentInputResponses])
+    } else {
+      // Or update existing item
+      const newList = [...listItems]
+      newList[selectedListItem] = currentInputResponses
+      setListItems(newList)
+    }
+    setCurrentInputResponses(resetCurrentResponses(inputFields))
+    setOpen(false)
+    setSelectedListItem(null)
+  }
+
+  const editItem = async (index: number) => {
+    if (!isEditable) return
+    setSelectedListItem(index)
+    setCurrentInputResponses(listItems[index])
+    setOpen(true)
+  }
+
+  const deleteItem = async (index: number) => {
+    setListItems(listItems.filter((_, i) => i !== index))
+    setOpen(false)
+    setCurrentInputResponses(resetCurrentResponses(inputFields))
+  }
+
+  const listDisplayProps: ListLayoutProps = {
+    listItems,
+    displayFormat,
+    editItem,
+    deleteItem,
+    fieldTitles: inputFields.map((e: TemplateElement) => e.title),
+    codes: inputFields.map((e: TemplateElement) => e.code),
+    Markdown,
+    isEditable,
+  }
+
+  const DisplayComponent =
```

Comment:
can this be component to export and use in SummaryView, it's a small tidy up request (since we are checking for const value rather then enum in SummaryView)

Reply:
Do you just mean make "DisplayComponent" a component and export that so that SummaryView can just render it without any checking logic?

Reply:
or use enum in the summaryView